packages/docusaurus-theme-classic/src/theme/DocCardList/index.tsx

```diff
@@ -21,15 +23,18 @@ function filterItems(items: PropSidebarItem[]): PropSidebarItem[] {
   });
 }
 
-export default function DocCardList({
-  items,
-}: {
-  items: PropSidebarItem[];
-}): JSX.Element {
+export default function DocCardList({items, isGenerated}: Props): JSX.Element {
+  const columnSize = 6;
+  const columnCount = 12 / columnSize;
   return (
     <div className="row">
       {filterItems(items).map((item, index) => (
-        <article key={index} className="col col--6 margin-bottom--lg">
+        <article
+          key={index}
+          className={clsx(`col col--${columnSize}`, {
+            'margin-bottom--lg': !isGenerated,
+            'margin-top--lg': isGenerated && index + 1 > columnCount,
```

Comment:
Not a very simple algorithm, maybe there is a better/shorter solution.
Examples:
- `isGenerated` -> docusaurus.io/docs/category/getting-started
- `!isGenerated` -> docusaurus.io/docs/advanced